### PR TITLE
Jae: Auto refresh/Check time and date when logging intangible time

### PR DESCRIPTION
**Pull request by RJBornToWin**
Bug description: Right now if someone has a poor internet connection and leaves the app open for a day and goes to log intangible time, the time will be logged using the date for the last time they were active in the app. I’d like the app to check/auto-refresh the time and date to make sure it is accurate when logging this intangible time.

Fix: Validate add time form upon submission and show corresponding error message. Admin can add submission for any date. Other users cannot. So the fix is limited to users other than admin. Any user can edit already entered form. So the fix is limited only to add. If incorrect date is populated and user tries to submit it, "Invalid date. Please refresh the page." is displayed as error message on the form.

Test cases:
1. Admin can add intangible time log for any date
2. User can add intangible time log only for current date (Need to access the app around 11.55 PM and leave it for 5 minutes), and try to add intangible time entry at 12.01 and see if incorrect date is populated. If incorrect then error message should be shown)

### CODE REVIEW

**Comment by one-community** [comment]
Wonderful! Thank you for putting PR raised by the bug on the bugs doc too. Please also always include the PR # and link. See how I did this there for this one as an example.

**Review by ShaunSully88 — APPROVED** [review]
Thanks a lot RJ, this is actually really helpful. Good to know I close in some areas, and grateful to learn the rest from you. Really appreciate the help on this, thank you!

File: src/components/Badge/Badge.jsx [hunk 1]
```diff
@@ -53,9 +53,9 @@ const Badge = props => {
     console.log('This is userId', userId);
```
Comment (EvianTan) [inline src/components/Badge/Badge.jsx:53]: No functional changes in this file.

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 2]
```diff
@@ -200,9 +200,15 @@ const TimeEntryForm = props => {
       result.dateOfWork = 'Date is required';
     } else {
       const date = moment(inputs.dateOfWork);
+      const today = moment(moment().tz('America/Los_Angeles').format('YYYY-MM-DD'));
       if (!date.isValid()) {
         result.dateOfWork = 'Invalid date';
       }
+      // Admin/Owner can add time entries for any dates. Other roles cannot. 
+      // Editing details of past date is possible for any role.
+      else if (currentUserRole !== 'Owner' && !edit && today.diff(date, 'days') != 0) {
```
Comment (RajasriJanaki) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:209]: In the code “Administrator” role is mentioned as owner. So I have checked only owner. Please correct me if I am wrong. Thanks.

Comment (RJBornToWin) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:209]: In the code “Administrator” role is mentioned as owner. So I have checked only owner. Please correct me if I am wrong. Thanks.

Comment (RJBornToWin) [reply src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx]: With the recent changes, Administrator role is handled as "Administrator" in the code. Updating the code to accommodate the same.

File: src/components/Badge/Badge.jsx [hunk 3]
```diff
@@ -53,9 +53,9 @@ const Badge = props => {
...
     let count = 0;
     if (props.userProfile.badgeCollection) {
-      console.log('userProfileBadgeCollectionId', props.userProfile._id);
+      // console.log('userProfileBadgeCollectionId', props.userProfile._id);
```
Comment (rauleffting) [inline src/components/Badge/Badge.jsx:56]: I think we can delete these debug outputs.

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 4]
```diff
@@ -386,7 +392,7 @@ const TimeEntryForm = props => {
     //Validation and variable initialization
     if (event) event.preventDefault();
     if (isSubmitting) return;
-    console.log('submitting');
+    // console.log('submitting');
```
Comment (rauleffting) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:395]: Here as well.

**Review by rauleffting — CHANGES_REQUESTED** [review]
Hi @RJBornToWin. Good job! The functionality is perfect, as you can see below:

https://user-images.githubusercontent.com/29555732/218766438-9bb069a2-fcff-4e67-a1e6-04949c09c76d.mp4

However, we still need to delete those debug outputs. Then, I'm going to approve! Thank you very much!

**Review by PrabhjotSembhi — COMMENTED** [review]
I am still able to add an intangible entry for the previous day. When time is @00:01 or more in the form it still  shows 16 Feb ( Previous date ) 
image



https://user-images.githubusercontent.com/82662768/219457611-d0aca459-e7d2-49a5-9b7d-3126669d864c.mp4

Please check video

**Comment by NyahJ** [comment]
Hi @RJBornToWin,

All your changes worked well, for the Admin and user accounts, they all worked as expected as you mentioned in the description. Thanks!

截屏2023-02-24 下午7 22 44
截屏2023-02-24 下午7 19 45

**Review by tingedudu8 — APPROVED** [review]
Hi @RJBornToWin, I tested your PR and the feature works properly for both volunteer and admin role. Thanks!
https://user-images.githubusercontent.com/70128933/221741991-3b9a797b-5a0e-4443-b797-2f1b6f49e140.mov

**Review by rauleffting — APPROVED** [review]
Hi @RJBornToWin. Changes worked perfectly. 

Approved!


https://user-images.githubusercontent.com/29555732/223794604-4d00556d-e387-4e17-b966-c59c7d58daa1.mp4

**Pull request merged**